src/lib/system-info.ts

```diff
@@ -85,7 +85,13 @@ export async function undervoltageDetected(): Promise<boolean> {
 	}
 }
 
-export async function getSysInfoToReport() {
+export async function getSysInfoToReport(
```

Comment:
I would remove this function and just construct the info object in the current-state file which sends the data. This way we can do...

```js
	let info = {
		is_undervolted: await sysInfo.undervoltageDetected()
	}
	if (SEND_METRICS){
		const metrics = await deviceMetrics()
		info = {
			metrics
			...info
		}
	}

	async function deviceMetrics() {
		const mem = await sysInfo.getMemoryInformation()
		const storage = await sysInfo.getStorageInfo()
		return {
			cpu_usage: await sysInfo.getCpuUsage(),
			memory_usage: mem.used,
			memory_total: mem.total,
			storage_usage: storage.storageUsed,
			storage_total: storage.storageTotal,
			storage_block_device: storage.blockDevice,
			cpu_temp: await sysInfo.getCpuTemp(),
			cpu_id: await sysInfo.getCpuId(),
		}
	}
```

Reply:
Removed 👍